core/src/main/java/com/linecorp/armeria/common/stream/StreamMessage.java

```diff
@@ -566,6 +566,28 @@ default EventExecutor defaultSubscriberExecutor() {
         return new AsyncMapStreamMessage<>(this, function);
     }
 
+    /**
+     * Transforms values emitted by this {@link StreamMessage} by applying the specified asynchronous
+     * {@link Function} and emitting the value the future completes with.
+     * The {@link StreamMessage} publishes items eagerly in the order that the futures complete.
+     * It does not necessarily preserve the order of the original stream.
+     * As per
+     * <a href="https://github.com/reactive-streams/reactive-streams-jvm#2.13">
+     * Reactive Streams Specification 2.13</a>, the specified {@link Function} should not return
+     * a {@code null} value nor a future which completes with a {@code null} value.
+     *
+     * <p>Example:<pre>{@code
+     * StreamMessage<Integer> streamMessage = StreamMessage.of(1, 2, 3, 4, 5);
+     * StreamMessage<Integer> transformed =
+     *     streamMessage.mapAsync(x -> CompletableFuture.completedFuture(x + 1));
+     * }</pre>
+     */
+    default <U> StreamMessage<U> mapAsyncParallel(
```

Comment:
`mapAsyncParallel` looks long to me.
How about using `mapParallel` that can also explain the behavior of this API?

Reply:
Should we give an option that limits max parallelism?
For example:
```java
// allow to concurrently execute up to 8
streamMessage.mapParallel(function, 8);
// unlimited
streamMessage.mapParallel(function);
```

Reply:
I think that's a good idea. In that case I can also rewrite `mapAsync` as `mapParallel(function, 1)`

Reply:
> I can also rewrite mapAsync as mapParallel(function, 1)

Good idea!